File: scripts/generate_pdf.js
import puppeteer from 'puppeteer';
import path from 'path';
import fs from 'fs';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

async function generatePDF() {
    console.log('Starting PDF generation...');
    const browser = await puppeteer.launch({
        args: ['--no-sandbox', '--disable-setuid-sandbox', '--disable-dev-shm-usage']
    });
    
    try {
        const page = await browser.newPage();
        
        // Load the local index.html
        const filePath = path.resolve(__dirname, '../index.html');
        const fileUrl = `file://${filePath}`;
        
        console.log(`Navigating to ${fileUrl}...`);
        await page.goto(fileUrl, { waitUntil: 'networkidle0' });
        
        // Create cv directory if it doesn't exist
        const cvDir = path.resolve(__dirname, '../cv');
        if (!fs.existsSync(cvDir)) {
            fs.mkdirSync(cvDir);
        }
        
        // Generate PDF with minimalist settings
        console.log('Generating PDF...');
        await page.pdf({
            path: path.resolve(cvDir, 'Pablo_Contreras_CV.pdf'),
            format: 'A4',
            printBackground: true,
            margin: {
                top: '0cm',
                right: '0cm',
                bottom: '0cm',
                left: '0cm'
            },
            displayHeaderFooter: false
        });

        console.log('PDF generated successfully at cv/Pablo_Contreras_CV.pdf');
    } catch (error) {
        console.error('Error during PDF generation:', error);
        throw error;
    } finally {
        await browser.close();
    }
}

generatePDF().catch(err => {
    console.error('Fatal error:', err);
    process.exit(1);
});

File: scripts/generate_pdf.py
import os
import asyncio
from pyppeteer import launch

async def generate_pdf():
    # Path to index.html
    current_dir = os.path.dirname(os.path.abspath(__file__))
    file_path = os.path.join(current_dir, '../index.html')
    file_url = f'file://{os.path.abspath(file_path)}'
    
    browser = await launch(args=['--no-sandbox'])
    page = await browser.newPage()
    
    await page.goto(file_url, {'waitUntil': 'networkidle0'})
    
    # Create cv directory
    cv_dir = os.path.join(current_dir, '../cv')
    if not os.path.exists(cv_dir):
        os.makedirs(cv_dir)
        
    pdf_path = os.path.join(cv_dir, 'Pablo_Contreras_CV.pdf')
    
    await page.pdf({
        'path': pdf_path,
        'format': 'A4',
        'printBackground': True,
        'margin': {
            'top': '0px',
            'right': '0px',
            'bottom': '0px',
            'left': '0px'
        }
    })
    
    await browser.close()
    print(f'PDF generated successfully at {pdf_path}')

if __name__ == '__main__':
    asyncio.get_event_loop().run_until_complete(generate_pdf())
